### PR TITLE
refactor: Move datalake to the new Atlas SDK

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Proposed changes

_Jira ticket:_ [INTMDB-1088](https://jira.mongodb.org/browse/INTMDB-1088)

Please include a summary of the fix/feature/change, including any relevant motivation and context.

Link to any related issue(s): 

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as
  expected)
- [ ] This change requires a documentation update

## Manual QA performed:

- [x] cfn invoke for each of CRUDL/cfn test
- [x] Updated resource in  [example](https://github.com/mongodb/mongodbatlas-cloudformation-resources/tree/master/examples)
- [x] Published to AWS private registry
- [x] Used the template in [example](https://github.com/mongodb/mongodbatlas-cloudformation-resources/tree/master/examples) to create and update a stack in AWS
- [x] Deleted stack to ensure resources are deleted
- [x] Created multiple resources in same stack
- [x] Validated in Atlas UI
- [x] Included screenshots

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code
- [x] For CFN Resources: I have released by changes in the private registry and proved by change
  works in Atlas

### CFN test
```bash
cfn test --function-name TestEntrypoint --verbose

==== 8 passed, 4 skipped, 12 deselected in 357.45s (0:05:57) ==
```

### CFN stack
Screenshot 2023-10-09 at 12 06 27
Screenshot 2023-10-09 at 12 07 03

### CODE REVIEW

File: cfn-resources/datalakes/cmd/resource/resource.go [hunk 1]
```diff
@@ -249,20 +237,20 @@ func List(req handler.Request, prevModel *Model, currentModel *Model) (handler.P
 		ResourceModels:  models,
 	}, nil
 }
-func flattenAWSBlock(awsConfig *mongodbatlas.CloudProviderConfig) *DataLakeAWSCloudProviderConfigView {
+func flattenAWSBlock(awsConfig *admin.DataLakeCloudProviderConfig) *DataLakeAWSCloudProviderConfigView {
 	if awsConfig == nil {
```
Comment (lantoli) [inline cfn-resources/datalakes/cmd/resource/resource.go:241]: can aws.AWSConfig also be nil?     and &awsConfig.AWSConfig.IAMUserARN, etc. ?

File: cfn-resources/datalakes/cmd/resource/resource.go [hunk 2]
```diff
@@ -272,98 +260,94 @@ func flattenDataLakeProcessRegion(processRegion *mongodbatlas.DataProcessRegion)
...
-func flattenDataLakeStorageDatabases(databases []mongodbatlas.DataLakeDatabase) []DataLakeDatabaseView {
+func flattenDataLakeStorageDatabases(databases []admin.DataLakeDatabaseInstance) []DataLakeDatabaseView {
 	database := make([]DataLakeDatabaseView, len(databases))
 	for ind := range databases {
 		database = append(database, DataLakeDatabaseView{
-			Name:                   &databases[ind].Name,
+			Name:                   databases[ind].Name,
 			Collections:            flattenDataLakeStorageDatabaseCollections(databases[ind].Collections),
 			Views:                  flattenDataLakeStorageDatabaseViews(databases[ind].Views),
-			MaxWildcardCollections: castNO64(databases[ind].MaxWildcardCollections),
```
Comment (lantoli) [inline cfn-resources/datalakes/cmd/resource/resource.go:282]: this looks like aws.IntValue or similar ones

File: cfn-resources/datalakes/cmd/resource/resource.go [hunk 3]
```diff
@@ -410,24 +394,20 @@ func validateProgress(client *mongodbatlas.Client, currentModel *Model, targetSt
...
+	dataLake, _, err := client.AtlasV2.DataFederationApi.GetFederatedDatabase(context.Background(), projectID, tenantName).Execute()
+	if err != nil || dataLake == nil {
 		return false
 	}
-	if dataLake != nil {
-		return true
-	}
 
-	return false
+	return true
```
Comment (lantoli) [inline cfn-resources/datalakes/cmd/resource/resource.go:405]: this is really:   return err == nil && dataLake != nil

File: cfn-resources/datalakes/cmd/resource/resource.go [hunk 4]
```diff
@@ -440,26 +420,26 @@ func dataLakeIsReady(client *mongodbatlas.Client, projectID, name, targetState s
 		}
 		return false, "", err
 	}
-	return dataLake.State == targetState, dataLake.State, nil
+	return *dataLake.State == targetState, *dataLake.State, nil
```
Comment (lantoli) [inline cfn-resources/datalakes/cmd/resource/resource.go:423]: can dataLake be nil so *dataLake will panic?

**Review by lantoli — DISMISSED** [review]

File: cfn-resources/datalakes/cmd/resource/resource.go (hunk 1 above)
Comment (andreaangiolillo) [reply cfn-resources/datalakes/cmd/resource/resource.go:241]: no, `AWSConfig` is required

File: cfn-resources/datalakes/cmd/resource/resource.go (hunk 4 above)
Comment (andreaangiolillo) [reply cfn-resources/datalakes/cmd/resource/resource.go]: I am not sure. I updated the logic to check for nill

File: cfn-resources/datalakes/cmd/resource/resource.go (hunk 3 above)
Comment (andreaangiolillo) [reply cfn-resources/datalakes/cmd/resource/resource.go]: updated

File: cfn-resources/datalakes/cmd/resource/resource.go (hunk 2 above)
Comment (andreaangiolillo) [reply cfn-resources/datalakes/cmd/resource/resource.go:282]: here I removed `castNO64` as it was not longer needed. Do I need to do anything as part of this comment?

**Review by lantoli — APPROVED** [review]

**Pull request merged**